MobiusCore/Source/Disposables/EmptyDisposable.swift

```diff
@@ -0,0 +1,30 @@
+// Copyright 2019-2024 Spotify AB.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// The `EmptyDisposable` class implements a `Disposable` type for when you don't have anything to dispose of.
+public final class EmptyDisposable: MobiusCore.Disposable {
+
+    /// Create an `EmptyDisposable`
+    public init() {}
+
+    public func dispose() {
+        // No-op
+    }
+}
+
+public extension Disposable where Self == EmptyDisposable {
+    static var empty: EmptyDisposable { EmptyDisposable() }
```

Comment:
Since the disposable will be most of the times used through this getter, I would suggest to add documentation to it, such as:

```suggestion
    /// Gets the disposable that does nothing when disposed.
    static var empty: EmptyDisposable { EmptyDisposable() }
```